### PR TITLE
doc: update deferred call handling doc with refactored APIs

### LINKED ISSUE #1376: Fusion: Handle Deferred Call Result document imporement
- the `import {EndpointRequest} from "./Connect";` is wrong, should be `import {EndpointRequest} from "@vaadin/flow-frontend";`
- In addition to the comments in the code example, could have some code in the code examples, e.g. how to use the call parameter, how to use the result, how to show a notification (could be just `console.log()`)
- Could have more explanation of the try-catch philosophy. Developers seem not getting the ideas until they play with a code for a while.
- During the DX testing, no developers found that the code example should be added to `index.ts`, should highlight that as least. Or maybe if we have the explanation of the try-catch philosophy, we can add why the code needs to be added to `index.ts`

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Fixes #1376

### CODE REVIEW

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc [hunk 1]
```diff
@@ -10,41 +10,47 @@ When deferring an endpoint call, its result is only available after the actual s
 
 By default, Vaadin sends the deferred calls without prompting or notifying the user, and their returned result is not available in the caller view. However, often applications still need to handle deferred call results, for example, to notify the user of sending the data collected during prior offline work.
 
-This article describes how to handle deferred call results on the client side.
+This article describes how to customise deferred call handling on the client side.
```
Comment (joheriks) [inline documentation/pwa/tutorial-handle-deferred-call-result.asciidoc:13]: `customize`, as docs use US English.

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc [hunk 2]
```diff
@@ -10,41 +10,47 @@ When deferring an endpoint call, its result is only available after the actual s
...
 
-== The `onDeferredCall` callback
+== The `deferredCallHandler` property
 
-Since the result is not available right away for deferred calls, Vaadin has `onDeferredCall` callback property on the `ConnectClient` TypeScript endpoint client class, which allows handling deferred call results.
+Vaadin has `deferredCallHandler` property on the `ConnectClient` TypeScript endpoint client class, which allows customising the deferred call handling.
```
Comment (joheriks) [inline documentation/pwa/tutorial-handle-deferred-call-result.asciidoc:17]: `customizing`

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc [hunk 3]
```diff
@@ -10,41 +10,47 @@ When deferring an endpoint call, its result is only available after the actual s
...
-    // The call will be removed from the deferred queue
-    // when sent successfully or if the error is handled here.
-
-    // Re-throw the error to keep the call in the queue:
-    // throw error;
+import { DeferredCallSubmitter } from '@vaadin/flow-frontend';
+
+client.deferredCallHandler = {
+  async handleDeferredCallSubmission(deferredCallSubmitter: DeferredCallSubmitter) {
+    try{
```
Comment (joheriks) [inline documentation/pwa/tutorial-handle-deferred-call-result.asciidoc:33]: Maybe reformat this with spaces between the keywords (`try`, `catch`) and `{`/`}`.

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc [hunk 4]
```diff
@@ -10,41 +10,47 @@ When deferring an endpoint call, its result is only available after the actual s
...
-== The `onDeferredCall` callback
+== The `deferredCallHandler` property
 
-Since the result is not available right away for deferred calls, Vaadin has `onDeferredCall` callback property on the `ConnectClient` TypeScript endpoint client class, which allows handling deferred call results.
+Vaadin has `deferredCallHandler` property on the `ConnectClient` TypeScript endpoint client class, which allows customising the deferred call handling. 
 
-To define a custom `onDeferredCall` callback, import the client and set the property in your `index.ts` file:
+The main philosophy for customization is to wrap the deferred call submission in a try-catch block, so that you can handle it correspondingly when a deferred call submission succeeds or fails.
+
+By default, Vaadin will remove the deferred call from the queue when submitted successfully or if the submission error is handled already, for example, in a try-catch. To keep the deferred call in the queue, you can call `deferredCallSubmitter.keepDeferredCallInTheQueue()` in the catch block.
```
Comment (joheriks) [inline documentation/pwa/tutorial-handle-deferred-call-result.asciidoc:21]: Slightly unclear when reading at this point what a `deferredCallSubmitter` is. Perhaps mention that it is a parameter received by the handler.

**Review by joheriks — CHANGES_REQUESTED** [review]

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc (hunk 1 above)
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-handle-deferred-call-result.asciidoc]: done

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc (hunk 2 above)
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-handle-deferred-call-result.asciidoc]: done

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc (hunk 3 above)
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-handle-deferred-call-result.asciidoc]: done

File: documentation/pwa/tutorial-handle-deferred-call-result.asciidoc (hunk 4 above)
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-handle-deferred-call-result.asciidoc]: good point, done

**Review by joheriks — APPROVED** [review]

**Pull request merged**